ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift

```diff
@@ -103,13 +111,22 @@ class PatchServerCodeTriggerTests: SmallTestBase {
             }
         }
     }
-    
-    func testPatchServerCodeTrigger_http_404() {
+    func testPatchServerCodeStateTrigger_http_404() {
+        let condition = Condition(clause: EqualsClause(field: "color", intValue: 0))
+        let predicate = StatePredicate(condition: condition, triggersWhen: TriggersWhen.CONDITION_FALSE_TO_TRUE)
+        patchServerCodeTrigger_http_404(predicate)
+
+    }
+    func testPatchServerCodeScheduleOnceTrigger_http_404() {
+        let predicate = ScheduleOncePredicate(scheduleAt: NSDate(timeIntervalSinceNow: 1000))
+        patchServerCodeTrigger_http_404(predicate)
+    }
+    func patchServerCodeTrigger_http_404(predicate: Predicate) {
         let setting:TestSetting = TestSetting()
         let api = setting.api
         let tag = "PatchServerCodeTriggerTests.testPatchServerCodeTrigger_http_404"
         let triggerID = "0267251d9d60-1858-5e11-3dc3-00f3f0b5"
-        let expectation = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404")
+        weak var expectation : XCTestExpectation! = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404_\(predicate.getEventSource().rawValue)")
```

Comment:
As far as i understood, people using weak attribute to avoid fulfill the expectation after the timeout.
But I don't think it's the best way since its hard to control when the expectation is nullified by ARC.
(In this point of view, I prefer to set nil in the completion handler explicitly instead of using weak)
I'm not very sure what was happend in the circleci build above but relation between adding weak and the failure is not very clear.

But OK. Let's separate the issue.